### PR TITLE
cmake: do not build debug_mutex or lockdep for Release build

### PR DESCRIPTION
**Pull request by tchaikov**
this change is a follow-up of #39905. it intends to reduce the

- compile time by not compiling debug mutex and lockdeps related stuff, and
- runtime memory footprint by removing unused lockdeps related global variables.

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by liewegas** [comment]
build issue aside, this looks good!

**Review by liewegas — APPROVED** [review]

File: src/common/lockdep.h [hunk 1]
```diff
@@ -30,4 +33,17 @@ extern int lockdep_will_lock(const char *n, int id, bool force_backtrace=false,
 extern int lockdep_locked(const char *n, int id, bool force_backtrace=false);
 extern int lockdep_will_unlock(const char *n, int id);
 extern int lockdep_dump_locks();
+
+#else
+
+static constexpr bool g_lockdep = false;
+static inline int lockdep_register(const char*) { return 0; }
+static inline void lockdep_unregister(int id) {}
+static inline int lockdep_will_lock(const char*, int id, bool, bool) { return  id; }
```
Comment (lxbsz) [inline src/common/lockdep.h:42]: This parameters should be the same with:

```
extern int lockdep_will_lock(const char *n, int id, bool force_backtrace=false,
			     bool recursive=false);
```

Or we are hitting the following errors:

```
../src/common/RWLock.h:106:70: error: too few arguments to function 'int lockdep_will_lock(const char*, int, bool, bool)'
  106 |     if (lockdep && g_lockdep) id = lockdep_will_lock(name.c_str(), id);
      |                                                                      ^
```

File: src/common/lockdep.h [hunk 2]
```diff
@@ -30,4 +33,17 @@ extern int lockdep_will_lock(const char *n, int id, bool force_backtrace=false,
...
+static inline int lockdep_locked(const char*, int id, bool) { return id; }
```
Comment (lxbsz) [inline src/common/lockdep.h:43]: The same here.

File: src/common/lockdep.h (hunk 1 above)
Comment (tchaikov) [reply src/common/lockdep.h]: @lxbsz fixed and repushed.

**Review by lxbsz — APPROVED** [review]

**Review by xxhdx1985126 — APPROVED** [review]

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-03-14_09:40:23-rados-wip-kefu-testing-2021-03-14-1612-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-03-15_02:29:57-rados-wip-kefu-testing-2021-03-15-0034-distro-basic-smithi/

**Comment by dillaman** [comment]
FYI --- this broke all the RBD gtests since we had a blanket `rados.conf_set("lockdep", "true")` call at test startup and now the tests are all failing. Can we ensure global changes like this hit all suites? I've been trying to track this down for hours since I haven't been able to repeat it locally. I'll fix on master but is this planned to be backported to pacific as well?

**Pull request merged**